### PR TITLE
Swift tests: RSC14d

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: ablySpec/RestClient.swift [hunk 1]
```diff
@@ -326,6 +327,82 @@ class RestClient: QuickSpec {
                 }
             }
 
+            // RSC14c
```
Comment (ricardopereira) [inline ablySpec/RestClient.swift]: Odd, I made this yesterday [RestClient 3cb5f8c](https://github.com/ably/ably-ios/commit/cfd4b3725a150a3a86744778d819c45115a3d2e8#diff-4ed1dc4a695e1026db6d7a9f33cb5f8c)! And I updated my branch with the latest merges.

Comment (ricardopereira) [inline ablySpec/RestClient.swift]: So dumb, sorry. The merge duplicated it.

**Comment by ricardopereira** [comment]
Current Swift tests state:
```
Executed 139 tests, with 51 failures (4 unexpected)
```
Most because of `RestClientStats.swift` and `RestChannel.swift`.

@mattheworiordan We need to merge it all to get the current state on this.

File: ablySpec/RestClient.swift [hunk 2]
```diff
@@ -326,6 +327,82 @@ class RestClient: QuickSpec {
...
+
+                            let rest = ARTRest(options: options)
+                            rest.httpExecutor = mockExecutor
+
+                            // Delay for token expiration
+                            delay(tokenParams.ttl) {
+                                // 40140 - token expired and will resend the request
+                                publishTestMessage(rest) { error in
+                                    expect(mockExecutor.responses.first?.allHeaderFields["X-Ably-ErrorCode"] as? String).to(equal("40140"))
+                                    expect(error).to(beNil())
```
Comment (mattheworiordan) [inline ablySpec/RestClient.swift]: Might be nice to check that the current token string is different to the previous one as well?

Comment (ricardopereira) [inline ablySpec/RestClient.swift]: @mattheworiordan Done. Thank you.

**Pull request closed without merging**